vault/src/system.rs

```diff
@@ -364,6 +362,23 @@ impl VaultIdManager {
         Ok(())
     }
 
+    async fn sweep_shared_wallet(&self) -> Result<(), Error> {
+        if self.btc_rpc_shared_wallet.get_pruned_height().await? == 0 {
+            // no need to sweep, full node can rescan
+            return Ok(());
+        } else if self.btc_rpc_shared_wallet_v2.get_last_sweep_height().await?.is_some() {
+            // already has sweep tx
+            return Ok(());
+        }
+
+        // sweep funds from shared wallet to shared-v2
+        let shared_v2_wallet_address = self.btc_rpc_shared_wallet_v2.get_new_sweep_address().await?;
```

Comment:
What will happen if we restart after we make the tx but before it gets included?

Reply:
I ran into this when testing yesterday, I've modified `get_last_sweep_height` to return an error if the sweep tx is still in the mempool